lib/langchain/conversation.rb

```diff
@@ -17,18 +17,18 @@ module Langchain
   #     end
   #
   class Conversation
-    attr_reader :options
+    attr_reader :options, :memory
 
     # Intialize Conversation with a LLM
     #
     # @param llm [Object] The LLM to use for the conversation
     # @param options [Hash] Options to pass to the LLM, like temperature, top_k, etc.
     # @return [Langchain::Conversation] The Langchain::Conversation instance
-    def initialize(llm:, **options, &block)
+    def initialize(llm:, memory_class: ConversationMemory, **options, &block)
```

Comment:
I'd pass a memory object instead of a class. So that the usage would be:
```ruby
llm = Langchain::LLM::OpenAI.new(...)
memory = MyCoolConversationMemory.new(llm: llm)
conversation = Langchain::Conversation(memory: memory)
```

Reply:
> memory = MyCoolConversationMemory.new(llm: llm)

What would this be needed for?

I think a better approach is:
```ruby
I'd pass a memory object instead of a class. So that the usage would be:

llm = Langchain::LLM::OpenAI.new(...)
memory = DatabaseConversationMemory.new(url: "postgres://...")
conversation = Langchain::Conversation(llm: llm, memory: memory)
```

Reply:
llm is used inside of ConversationMemory to calculate token limit and also to summarize history once the limit is reached. We can change that somehow, but that's not related to my comment. My comment is about passing an instance of memory to conversation instead of just class_name.